src/artifacts_poetry_plugin/plugin.py

```diff
@@ -9,143 +9,234 @@
+        )
+        repo_data = self.get_repository_data(config, repo_name, io)
+        self.upload_packages(
+            poetry_files=poetry_files,
+            url=repo_data.url,
+            username=repo_data.username,
+            password=repo_data.password,
+            io=io,
+        )
+        self.log_non_deployable_packages(poetry_files)
+
+
+class ArtifactsDeployCommand(Command):
     name = ARTIFACTS_DEPLOY
 
     arguments = [
-        argument(REPOSITORY_NAME, "The name of the repository to deploy artifacts")
+        argument(
+            REPOSITORY_NAME,
+            "The name of the repository to which artifacts should be deployed",
+        )
     ]
 
-    help = "The Artifacts Deploy command is used to deploy all of a Python project's dependencies to a repository"
+    help = "The Artifacts Deploy command is used to deploy a Python project's dependencies to a repository"
 
     def handle(self) -> int:
         self.line("Artifacts Deploy")
         repo_name = self.argument(REPOSITORY_NAME)
-        poetry = Factory().create_poetry(Path(os.getcwd()))
-        config = poetry.config
-        repo_data = get_repository_data(config, repo_name, self.io)
-        url = repo_data["url"]
-        username = repo_data["username"]
-        password = repo_data["password"]
-        packages, non_deployable_packages = get_deploy_packages(
-            poetry,
-            poetry.config.artifacts_cache_directory.absolute().as_posix(),
-        )
-        upload_packages(
-            poetry,
-            packages=packages,
-            url=url,
-            username=username,
-            password=password,
-            io=self.io,
+        artifacts_deploy_helper = ArtifactsDeployHelper()
+        artifacts_deploy_helper.deploy_poetry(repo_name, False, self.io)
+        return 0
+
+
+class ArtifactsDeployAllCommand(Command):
+    name = ARTIFACTS_DEPLOY_ALL
+
+    arguments = [
+        argument(
+            REPOSITORY_NAME,
+            "The name of the repository to which artifacts should be deployed",
         )
-        for package in non_deployable_packages:
-            logger.warn(
-                f"Could not find any wheel files for package {package.name}. Package may be statically linked or doesnt exist in artifacts cache"
-            )
+    ]
+
+    help = "The Artifacts Deploy command is used to deploy all downstream Python project's dependencies to a repository"
+
+    def handle(self) -> int:
+        self.line("Artifacts Deploy All")
+        repo_name = self.argument(REPOSITORY_NAME)
+        artifacts_deploy_helper = ArtifactsDeployHelper()
+        artifacts_deploy_helper.deploy_poetry(repo_name, True, self.io)
         return 0
 
 
-def factory():
-    return ArtifactsDeploy()
+def artifacts_deploy_all_factory():
+    return ArtifactsDeployAllCommand()
+
+
+def artifacts_deploy_factory():
+    return ArtifactsDeployCommand()
 
 
 class ArtifactsPoetryPlugin(poetry.plugins.application_plugin.ApplicationPlugin):
```

Comment:
Would it make sense to put some of these classes in their own dedicated files for organizational purposes? Unless the are all defined in plugin.py since they're only used in this file